UnitySDK/Assets/ML-Agents/Scripts/Agent.cs

```diff
@@ -542,6 +545,24 @@ public void InitializeSensors()
             m_VectorSensorBuffer = new float[numFloatObservations];
         }
 
+        void InitializeRewardProvider()
```

Comment:
InitializeLegacyRewardProvider ?
Why does `LegacyRewardProviderComponent` exist ? Isn't the legacy one always a part of the Agent.
Also, from our conversation, I do not think this reward provider will be a legacy, it is too useful to call AddReward on a reward provider. I would try to make it a full feature.

Reply:
Renamed to LowLevelRewardProvider.  The RewardProviderComponent is there for ease of use.  Users can add components to objects.  If the user decides to add a different reward provider, then the LowLevelRewardProvider doesn't need to exist.  Unless you're thinking that all reward providers inherit from the LowLevelRewardProvider.  I didn't think that's what you meant when we were talking.